File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/mapper/SortOrderDataMapper.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.mapper;

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.model.SortOrder;
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.model.SortOrderModel;

import javax.inject.Inject;

class SortOrderDataMapper {

    @Inject
    public SortOrderDataMapper() {
    }

    public SortOrder modelToDomain(SortOrderModel sortOrderModel) {
        switch (sortOrderModel) {
            case TITLE:
                return SortOrder.TITLE;
            case NO_OF_EPISODES:
                return SortOrder.NO_OF_EPISODES;
        }
        return null;
    }
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/mapper/SortOrderDataMapper.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.mapper

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.model.SortOrder
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.model.SortOrderModel

import javax.inject.Inject

internal class SortOrderDataMapper @Inject
constructor() {

    fun modelToDomain(sortOrderModel: SortOrderModel): SortOrder? {
        return when (sortOrderModel) {
            SortOrderModel.TITLE -> SortOrder.TITLE
            SortOrderModel.NO_OF_EPISODES -> SortOrder.NO_OF_EPISODES
        }
        return null
    }
}
